src/allocation_engine/interval_tree.rs

```diff
@@ -33,14 +37,139 @@ pub struct Constraint {
     pub policy: AllocPolicy,
 }
 
-/// A closed interval range [min, max] used to describe a
+/// A closed interval range [range_start, range_end] used to describe a
 /// memory slot that will be assigned to a device by the VMM.
 /// This structure represents the key of the Node object in
 /// the interval tree implementation.
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Hash, Ord)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Hash, Ord, Debug)]
 pub struct Range {
-    pub min: u64,
-    pub max: u64,
+    /// Lower boundary of the interval.
+    range_start: u64,
+    /// Upper boundary of the interval.
+    range_end: u64,
+}
+
+impl Range {
+    /// Create a new Range object.
+    pub fn new(range_start: u64, range_end: u64) -> Result<Self> {
+        if range_start > range_end {
+            return Err(Error::InvalidRange(range_start, range_end));
+        }
+        Ok(Range {
+            range_start,
+            range_end,
+        })
+    }
+
+    /// Get length of the range.
+    pub fn length(&self) -> u64 {
```

Comment:
`len()` is probably more idiomatic.

Reply:
Clippy is gonna complain if we have defined len but not `is_empty` and in this context is_empty doesn't have sense in my opinion.

Reply:
@AlexandruCihodaru you can also ignore the clippy complaint with `#[allow(clippy::len_without_is_empty)]` above the `impl Range {`.